datafusion/physical-expr/src/aggregate/correlation.rs

```diff
@@ -151,6 +151,13 @@ impl Accumulator for CorrelationAccumulator {
         Ok(())
     }
 
+    fn retract_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
+        self.covar.retract_batch(values)?;
+        self.stddev1.retract_batch(&[values[0].clone()])?;
```

Comment:
I think you can avoid these clones with something like

```rust
        self.stddev1.retract_batch(&values[0..1])?;
        self.stddev2.retract_batch(&values[1..2])?;
```

(aka slice the slice!)